Yoast/yoast-components
# Stories/9834 ditch snippet variables

## Summary

This PR can be summarized in the following changelog entry:

* Removes certain replacement variables from appearing in the suggestions in the snippet editor (`currentdate`, `currenttime`, `currentyear`, `currentmonth`, `currentday`, `userid`, `searchphrase`, and `archive_title`)

## Relevant technical choices:

* Introduced a new prop which can contain the names of any replacement variables that should be removed from the suggestions
* This allows specific inclusion/exclusion of replacement variables depending on the instance of the replacement variable editor (the upcoming implementation for the settings page can include `searchphrase` for example)
* This also allows the replacement variables to be used as normal regardless of whether or not they appear in the suggestions menu

## Test instructions

This PR can be tested by following these steps:

* Link wordpress-seo to yoast-components and build
* When editing a snippet, bring up the replacement variable suggestions in the title or meta description fields
* Make sure that the above mentioned replacement variables are not listed in the suggestions 
* Make sure that's the case for both pages and posts
* Typing any of the above replacement variables in their full form (e.g. %%currentdate%%) will correctly display the value (but not bring it up in the suggestions) 

Fixes https://github.com/Yoast/wordpress-seo/issues/9834
